src/test/java/ch/wisv/areafiftylan/products/service/TicketServiceTest.java

```diff
@@ -0,0 +1,664 @@
+        thrown.expectMessage("Ticket not found");
+        ticketService.removeTicket(id);
+    }
+
+    @Test
+    public void removeTicketNull() {
+        thrown.expect(TicketNotFoundException.class);
+        thrown.expectMessage("Ticket not found");
+        ticketService.removeTicket(null);
+    }
+
+    @Test
+    public void getNumberSoldOfTypeZero() {
+        Ticket ticket1 = persistTicket();
+        TicketType ticketType = ticket1.getType();
+        ticketService.removeTicket(ticket1.getId());
+        assertEquals(Integer.valueOf(0), ticketService.getNumberSoldOfType(ticketType));
+    }
+
+    @Test
+    public void getNumberSoldOfTypeOne() {
+        Ticket ticket1 = persistTicket();
+        TicketType ticketType = ticket1.getType();
+        assertEquals(Integer.valueOf(1), ticketService.getNumberSoldOfType(ticketType));
+    }
+
+    @Test
+    public void getNumberSoldOfTypeAdd() {
+        Ticket ticket1 = persistTicket();
+        TicketType ticketType = ticket1.getType();
+        assertEquals(Integer.valueOf(1), ticketService.getNumberSoldOfType(ticketType));
+        persistTicket();
+        assertEquals(Integer.valueOf(2), ticketService.getNumberSoldOfType(ticketType));
+    }
+
+    @Test
+    public void getNumberSoldOfTypeRemove() {
+        Ticket ticket1 = persistTicket();
+        TicketType ticketType = ticket1.getType();
+        persistTicket();
+        assertEquals(Integer.valueOf(2), ticketService.getNumberSoldOfType(ticketType));
+        ticketService.removeTicket(ticket1.getId());
+        assertEquals(Integer.valueOf(1), ticketService.getNumberSoldOfType(ticketType));
+    }
+
+    @Test
+    public void getNumberSoldOfTypeTypeNull() {
+        assertEquals(Integer.valueOf(0), ticketService.getNumberSoldOfType(null));
+    }
+
+    @Test
+    public void findValidTicketsByOwnerUsername() {
+        //TODO I can't get this method to do my bidding...
+        /*
+        User user = persistUser();
+        Ticket ticket = ticketService.requestTicketOfType(
+        user, TEST_TICKET, Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION));
+        testEntityManager.persist(ticket);
+        Collection<Ticket> tickets = ticketService.findValidTicketsByOwnerUsername(user.getUsername());
+        assertEquals(Collections.singleton(ticket), tickets);
+        */
+    }
+
+    @Test
+    public void getAllTicketsZero() {
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(0, tickets.size());
+    }
+
+    @Test
+    public void getAllTicketsOne() {
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(1, tickets.size());
+    }
+
+    @Test
+    public void getAllTicketsTwo() {
+        persistTicket();
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(2, tickets.size());
+    }
+
+    @Test
+    public void validateTicketFromInvalid() {
```

Comment:
Test name could be changed to `validateTicket`, this is just the 'base' test. A more thorough test can be given a more specific name.